### PR TITLE
fix(prost-build): Remove `derived(Copy)` on boxed fields

### PR DESCRIPTION
**Pull request by ldm0**
Fix one of the problems I found while rebasing #705: boxed fields are not `Copy`

### CODE REVIEW

**Review by caspermeijn — CHANGES_REQUESTED** [review]
So, basically `Config::boxed()` is broken in the latest released version because in many circumstances it will derive copy on the manually marked boxed field.

I would like to see a new test that prevents this in the future.

File: prost-build/src/message_graph.rs [hunk 1]
```diff
@@ -87,50 +92,4 @@ impl MessageGraph {
 
         has_path_connecting(&self.graph, outer, inner, None)
     }
-
-    /// Returns `true` if this message can automatically derive Copy trait.
-    pub fn can_message_derive_copy(&self, fq_message_name: &str) -> bool {
```
[the remark this answers is not in the document]
Comment (ldm0) [reply prost-build/src/message_graph.rs:92]: > Currently, if two messages include each other, the automatic boxing happens at both sides of the recursive include. If one side is already manually boxed, then it doesn't have to be at the other side.

That's interesting. I created another feature PR to address this: https://github.com/tokio-rs/prost/pull/1160

**Comment by ldm0** [comment]
> I would like to see a new test that prevents this in the future.

Test added.

**Review by caspermeijn — APPROVED** [review]
Thank you for your contribution

**Pull request merged**